File: Samples/Verbum/how-to-implement-the-insert-document-field-dialog-winforms-version-t223036-14.2.6-/CS/WindowsFormsApplication1/InsertFieldModules/ucDOCVARIABLE.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace WindowsFormsApplication1.InsertFieldModules {
    public partial class ucDOCVARIABLE : UserControl, IFieldParametersDialog {
        public ucDOCVARIABLE() {
            InitializeComponent();
        }

        #region IFieldParametersDialog Members

        public string GenerateFieldParametersString() {
            string resultString = "";
            if(textEditVariableName.Text.Trim() == "") return InsertFieldRichEditHelper.EmptyFieldCode;

            resultString += textEditVariableName.Text.Trim();
            if(textEditArgument1.Text.Trim() != "")
                resultString += " " + textEditArgument1.Text.Trim();
            if(textEditArgument2.Text.Trim() != "")
                resultString += " " + textEditArgument2.Text.Trim();
            if(textEditArgument3.Text.Trim() != "")
                resultString += " " + textEditArgument3.Text.Trim();
            if(textEditArgument4.Text.Trim() != "")
                resultString += " " + textEditArgument4.Text.Trim();
            return resultString;
        }

        #endregion
    }
}

File: Samples/Verbum/how-to-implement-the-insert-document-field-dialog-winforms-version-t223036-14.2.6-/VB/WindowsFormsApplication1/InsertFieldModules/ucDOCVARIABLE.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms

Namespace WindowsFormsApplication1.InsertFieldModules
	Partial Public Class ucDOCVARIABLE
		Inherits UserControl
		Implements IFieldParametersDialog
		Public Sub New()
			InitializeComponent()
		End Sub

		#Region "IFieldParametersDialog Members"

		Public Function GenerateFieldParametersString() As String Implements IFieldParametersDialog.GenerateFieldParametersString
			Dim resultString As String = ""
			If textEditVariableName.Text.Trim() = "" Then
				Return InsertFieldRichEditHelper.EmptyFieldCode
			End If

			resultString &= textEditVariableName.Text.Trim()
			If textEditArgument1.Text.Trim() <> "" Then
				resultString &= " " & textEditArgument1.Text.Trim()
			End If
			If textEditArgument2.Text.Trim() <> "" Then
				resultString &= " " & textEditArgument2.Text.Trim()
			End If
			If textEditArgument3.Text.Trim() <> "" Then
				resultString &= " " & textEditArgument3.Text.Trim()
			End If
			If textEditArgument4.Text.Trim() <> "" Then
				resultString &= " " & textEditArgument4.Text.Trim()
			End If
			Return resultString
		End Function

		#End Region
	End Class
End Namespace
